storage.py

```diff
@@ -0,0 +1,235 @@
+import contextlib
+
+from pymemcache.client.base import Client
+from pymemcache.serde import (python_memcache_serializer,
+                              python_memcache_deserializer)
+
+from clcache import CacheFileStrategy, getStringHash, printTraceStatement, CompilerArtifacts, \
+    CACHE_COMPILER_OUTPUT_STORAGE_CODEC
+
+
+class CacheDummyLock(object):
+    def __enter__(self):
+        pass
+
+    def __exit__(self, typ, value, traceback):
+        pass
+
+
+class CacheMemcacheStrategy(object):
+    def __init__(self, server, cacheDirectory=None, manifestPrefix='manifests_', objectPrefix='objects_'):
+        self.fileStrategy = CacheFileStrategy(cacheDirectory=cacheDirectory)
+        # XX Memcache Strategy should be independent
```

Comment:
- I would like to make file strategy and memcache strategy independent. For that I would need to change how `set/getEntry` work. ATM the local cache is used as intermediary, so memcache always 'pollutes' the local cache 
`        cachedArtifacts = cache.getEntry(cachekey)  # memcache puts objectFile into local cache
         copyOrLink(cachedArtifacts.objectFilePath, objectFile)  # copy local cache to build folder`